### DISCUSSION TITLE
[Question] Why isn't everything here just put on neofetch?

### DISCUSSION THREAD

**Discussion by ShawkMusic** [fastfetch-cli/fastfetch#348] [Q&A] [answered]
If fastfetch is so much faster then neofetch and they're both open source and MIT licensed, why isn't the code here just put on neofetch? What's the point of an entirely new program to do the same thing but better? Why not just make the original neofetch faster? It's not like there's people out there who specifically need neofetch to be slow

**Comment by hmlendea** [comment]
Because this would require 100% of neofetch's code to be wiped, and replaced with this code.

Neofetch is written in bash, fastfetch is written in C.

So it would not be just an improvement, it would mean killing neofetch to put fastfetch in its place... or some might say it would basically be stealing its name.

PS: It's also worth mentioning that the chosen language alone would be reason enough for some people to chose one or the other. Why? I don't know, but some people seem to be very religious about coding languages, for some reason.
Pitchforks and torches would be raised if anyone dared to change it to another language... and a dozen more *fetch forks / reimplementations would be spawned out of that conflict.

**Comment by LinusDierheimer** [comment, marked as answer]
A second point is that fastfetch and neofetch target a different set of machines. Fastfetch targets only Linux Systems, so i can make use of a lot of linux specific features to achieve the speed fastfetch has. Neofetch also supports MacOS, Windows and a lot of other different systems. If you wanted fastfetch to also support those systems, you basically would have to rewrite the whole thing from scratch.

That is a common deal in software development: If you support less, you can support the remaining better.
